src/firstboot/firstboot.c

```diff
@@ -1040,6 +1042,69 @@ static int process_kernel_cmdline(int rfd) {
         return 0;
 }
 
+static int process_reset(int rfd) {
+        _cleanup_free_ char *esp_path = NULL, *random_seed = NULL;
+        _cleanup_close_ int esp_fd = -EBADF;
+        int r;
+
+        if (!arg_reset)
+                return 0;
+
+        FOREACH_STRING(s,
+                       "etc/locale.conf",
+                       "etc/vconsole.conf",
+                       "etc/localtime",
```

Comment:
my educated guess is that people might not like it if we blanket remove their pre-defined timezones and locale settings.

i guess we sooner or later will need `--reset-locale`, `--reset-timezone` flags, that condition that, similar to the existing `--prompt-locale`, `--prompt-keymap`, `--prompt-…` flags...